components/commercial_filter/filters/location/location_filter_view.coffee

```diff
@@ -26,12 +27,21 @@ module.exports = class LocationFilterView extends Backbone.View
   bloodHound: ->
     cities = _.pluck GEOCODED_CITIES, 'name'
     @hound = new Bloodhound
-      datumTokenizer: Bloodhound.tokenizers.whitespace,
-      queryTokenizer: Bloodhound.tokenizers.whitespace,
-      local: cities
+      datumTokenizer: Bloodhound.tokenizers.obj.whitespace('value'),
+      queryTokenizer: @queryTokenizer,
+      local: _.map(cities, (city) => { value: @translate(city), display: city })
```

Comment:
return the `value` which gets used in the tokenizers and has the accented characters removed, but also return the unmodified `display` to render.